src/store/api/worldUsers.ts

```diff
@@ -0,0 +1,112 @@
+import { createApi, fakeBaseQuery } from "@reduxjs/toolkit/query/react";
+import firebase from "firebase/app";
+
+export interface WorldUsersApiArgs {
+  relatedLocationIds: string[];
+}
+
+// TODO: define this properly
+export interface WorldUsersData {
+  userIds?: string[];
+}
+
+// TODO: https://redux-toolkit.js.org/rtk-query/api/createApi
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/customizing-queries
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#streaming-data-with-no-initial-request
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/migrating-to-rtk-query
+export const worldUsersApi = createApi({
+  reducerPath: "worldUsersApi",
+  baseQuery: fakeBaseQuery(),
+  endpoints: (builder) => ({
+    worldUsers: builder.query<WorldUsersData, WorldUsersApiArgs>({
+      // The query is not relevant here as the data will be provided via streaming updates.
+      // A queryFn returning an empty array is used, with contents being populated via
+      // streaming updates below as they are received.
+      queryFn: () => ({ data: {} }),
+      async onCacheEntryAdded(
+        { relatedLocationIds },
+        { updateCachedData, cacheEntryRemoved }
+      ) {
+        // TODO: use api/* layer for this?
+        const usersCollectionRef = firebase.firestore().collection("users");
+
+        const worldUsersQuery = usersCollectionRef.where(
+          "enteredVenueIds",
+          "array-contains-any",
+          relatedLocationIds
+        );
+
+        const unsubscribeListener = worldUsersQuery.onSnapshot((snapshot) => {
```

Comment:
[note] This `worldUsersApi` is where all of the 'connect to firestore snapshot listeners'/etc stuff is happening now. It makes use of redux toolkit query to abstract/encapsulate a lot of the boilerplate/etc we would otherwise need to write.